sos/plugins/kubernetes.py

```diff
@@ -15,48 +15,128 @@
+from os import path
 
 
 class kubernetes(Plugin, RedHatPlugin):
 
     """Kubernetes plugin
     """
 
-    option_list = [("podslog", "capture logs for pods", 'slow', False)]
+    # Red Hat Atomic Platform and OpenShift Enterprise use the
+    # atomic-openshift-master package to provide kubernetes
+    packages = ('kubernetes', 'kubernetes-master', 'atomic-openshift-master')
+    files = ("/etc/origin/master/master-config.yaml",)
+
+    option_list = [
+        ("all", "also collect --all-namespaces output separately",
+            'fast', True),
+        ("describe", "capture descriptions of all kube resources",
+            'fast', True),
+        ("podlogs", "capture logs for pods", 'slow', False),
+    ]
+
+    def check_is_master(self):
+        if any([
+            path.exists("/var/run/kubernetes/apiserver.key"),
+            path.exists("/etc/origin/master/master-config.yaml")
+        ]):
+            return True
+        return False
 
     def setup(self):
         self.add_copy_spec("/etc/kubernetes")
         self.add_copy_spec("/var/run/flannel")
 
-        # Kubernetes master info
-        self.add_cmd_output("kubectl version")
-        self.add_cmd_output("kubectl get -o json pods")
-        self.add_cmd_output("kubectl get -o json nodes")
-        self.add_cmd_output("kubectl get -o json services")
-        self.add_cmd_output("kubectl get -o json replicationController")
-        self.add_cmd_output("kubectl get -o json events")
-
-        # This could use a single call:
-        #
-        #   add_journal(units=["kubelet", "kube-apiserver", ... ])
-        #
-        # But this would merge all units into a single text stream - to
-        # preserve existing file layout in archives keep these as
-        # separate journalctl calls.
-        self.add_journal(units="kubelet")
-        self.add_journal(units="kube-apiserver")
-        self.add_journal(units="kube-controller-manager")
-        self.add_journal(units="kube-scheduler")
-        self.add_journal(units="kube-proxy")
-
-        if self.get_option('podslog'):
-            result = self.get_command_output("kubectl get pods")
-            if result['status'] == 0:
-                for line in result['output'].splitlines()[1:]:
-                    pod_name = line.split(" ")[0]
-                    self.add_cmd_output([
-                        "{0} log {1}".format("kubectl", pod_name)
-                    ])
+        svcs = [
+            'kubelet',
+            'kube-apiserver',
+            'kube-proxy',
+            'kube-scheduler',
+            'kube-controller-manager'
+        ]
+
+        for svc in svcs:
+            self.add_journal(units=svc)
+
+        # We can only grab kubectl output from the master
+        if self.check_is_master():
+            kube_cmd = "kubectl "
+            if path.exists('/etc/origin/master/admin.kubeconfig'):
+                kube_cmd += "--config=/etc/origin/master/admin.kubeconfig"
+
+            kube_get_cmd = "get -o json "
```

Comment:
Why is this variable set here if it is not used in the current block (separated by blank lines)? Shouldnt it be rather on previous line where commands/arguments are to be specified?